extensions/librdkafka/PublishKafka.cpp

```diff
@@ -925,12 +931,10 @@ void PublishKafka::onTrigger(const std::shared_ptr<core::ProcessContext> &contex
     }
 
     std::string kafkaKey;
-    kafkaKey = "";
-    if (context->getDynamicProperty(MessageKeyField, kafkaKey, flowFile) && !kafkaKey.empty()) {
-      logger_->log_debug("PublishKafka: Message Key Field [%s]", kafkaKey);
-    } else {
+    if (!context->getProperty(KafkaKey, kafkaKey, flowFile) || kafkaKey.empty()) {
```

Comment:
if `MessageKeyField` worked before and now it no longer works we should remove it altogether and not mark it deprecated, if we would like to enable a "soft transition" we should mark it deprecated but further support the feature and produce loud warnings that it will not be so for long

Reply:
It's complicated.  `MessageKeyField` used to "work" in the sense that if you set it (e.g. to `kafka.message.key` or `${kafka.message.key}`) then every message sent to Kafka would have the message key set to this literal value ("kafka.message.key" or "${kafka.message.key}") without any substitution, i.e. all messages would have the same key.

The description of the `MessageKeyField` property was "The name of a field in the Input Records that should be used as the Key for the Kafka message. Supports Expression Language: true (will be evaluated using flow file attributes)", every part of which is false.

I hope nobody was using this, but I think that silently changing the message keys to the flow file UUID (which is what will happen if you were using `MessageKeyField` previously and don't change anything) is better than keeping the old behavior where every message had the same key.  Message keys are supposed to be unique.

What do you think of keeping the behavior as it is, but adding an error log if `MessageKeyField` is set?

Reply:
I don't think this ever worked since we try to get the `MessageKeyField` as if it were a dynamic property (`context->getDynamicProperty`)

Reply:
does an exception in an onSchedule really prevents a MiNiFi startup? that would be really bad, as there is no way to fix such a malformed flow through a c2 server

Reply:
your approach of keeping it as is, and logging errors if `MessageKeyField` is set, could be our best bet, given the constraints

Reply:
You're right about both MiNiFi startup and the old (non-)behavior or `MessageKeyField`.

I have added an error log in `onSchedule()` when `MessageKeyField` is set.